File: backend_golang/api/orders/usecase/ucase.go
package usecase

import (
	"backend_golang/api/orders"
	"backend_golang/domain/models"
	"context"
)

type ordersUsecase struct {
	repository orders.Repository
}

// NewOrdersUsecase ...
func NewOrdersUsecase(repository orders.Repository) orders.Usecase {
	return &ordersUsecase{
		repository: repository,
	}
}

// All ...
func (u *ordersUsecase) All(simpleContext context.Context) (*models.Store, error) {
	return u.repository.Find(simpleContext)
}

File: backend/api/orders/usecase/ucase.py
from .interface import OrdersUsecaseBlueprint
from ..repository.interface import OrdersCollectionBlueprint

class OrdersUsecase(OrdersUsecaseBlueprint):
  repository: OrdersCollectionBlueprint | None = None
  
  def __init__(self, repo: OrdersCollectionBlueprint):
    self.repository = repo
  
  def All(self):
    return self.repository.find()
